src/Frames/Joins.hs

```diff
@@ -0,0 +1,191 @@
+{-# LANGUAGE ConstraintKinds, DataKinds, FlexibleContexts, FlexibleInstances,
+             KindSignatures, MultiParamTypeClasses, PolyKinds,
+             ScopedTypeVariables, TypeFamilies, TypeOperators,
+             UndecidableInstances, TemplateHaskell, QuasiQuotes,
+             Rank2Types #-}
+module Frames.Joins where
+import Data.Discrimination
+import Data.Functor.Contravariant
+import Data.Foldable as F
+import Frames
+import Frames.Melt (RDeleteAll)
+import Frames.InCore (RecVec)
+import Data.Vinyl.TypeLevel
+import Data.Vinyl
+import Data.Vinyl.Functor
+import Data.Functor.Contravariant.Divisible
+import qualified Data.Text as T
+
+dropCols :: 
+  (rs' ~ RDeleteAll fs rs
+  , rs' ⊆ rs) =>
+  proxy fs -> Record rs -> Record rs'
+dropCols _ = rcast 
+
+mergeRec :: 
+  (fs ⊆ rs2
+  , rs2' ⊆ rs2
+  , rs2' ~ RDeleteAll fs rs2
+  , rs ⊆ (rs ++ rs2')) =>
+  proxy fs ->
+  Record rs ->
+  Record rs2 ->
+  Record (rs ++ rs2')
+mergeRec cols rec1 rec2 =
+  rec1 <+> rec2'
+  where
+    rec2' = dropCols cols rec2 
+
+instance (AllCols Grouping rs
+         , Grouping (Record rs)
+         , Grouping (s :-> r)
+         , Grouping r
+         ) =>
+         Grouping (Record ((s :-> r) : rs)) where
+  grouping = divide recUncons grouping grouping
+
+instance Grouping (Record '[]) where
+  grouping = conquer
+
+instance (Grouping a) => Grouping (s :-> a) where
+   grouping = contramap getCol grouping
+
+instance Grouping Text where
+  grouping = contramap T.unpack grouping
+
+-- | Perform an inner join operation on two frames
+-- matching on 
```

Comment:
I think this comment was accidentally truncated at some point.

Reply:
oops, will fix.